src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSMessagingUtils.cs

```diff
@@ -0,0 +1,66 @@
+// <copyright file="AWSMessagingUtils.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.Contrib.Instrumentation.AWS.Implementation;
+
+internal static class AWSMessagingUtils
+{
+    // SQS/SNS message attributes collection size limit according to
+    // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html and
+    // https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html
+    private const int MaxMessageAttributes = 10;
+
+    internal static void Inject(IRequestContextAdapter requestAdapter, PropagationContext propagationContext)
+    {
+        if (!requestAdapter.HasMessageBody ||
+            !requestAdapter.HasOriginalRequest)
+        {
+            return;
+        }
+
+        var carrier = new Dictionary<string, string>();
+        Propagators.DefaultTextMapPropagator.Inject(propagationContext, carrier, Setter);
+
+        int attributesCount = requestAdapter.AttributesCount;
+        if (carrier.Count + attributesCount > MaxMessageAttributes)
+        {
+            // TODO: Add logging (event source).
+            return;
+        }
+
+        int nextAttributeIndex = attributesCount + 1;
+        foreach (var param in carrier)
+        {
+            if (requestAdapter.ContainsAttribute(param.Key))
+            {
+                continue;
```

Comment:
If any attribute is already contained, we should skip injection entirely and not inject any of the attributes. Partial injection should be avoided (e.g. we might end-up injecting a tracestate that is unrelated to an already existing traceparent)

(Alternatively, we could overwrite existing attributes, I think that's the principle most other instrumentations follow, though I am not sure it would be appropriate here)

See also my original comment https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1051#discussion_r1119857473

Reply:
Implemented logic: if at least one attribute is already present in the request then we skip the injection.

Reply:
Isn't the `if` above now redundant?